worker/src/RTC/SctpAssociation.cpp

```diff
@@ -115,29 +115,25 @@ namespace RTC
 	  size_t maxSctpMessageSize,
 	  size_t sctpSendBufferSize,
 	  bool isDataChannel)
-	  : listener(listener), os(os), mis(mis), maxSctpMessageSize(maxSctpMessageSize),
-	    sctpSendBufferSize(sctpSendBufferSize), isDataChannel(isDataChannel)
+	  : id(DepUsrSCTP::GetNextSctpAssociationId()), listener(listener), os(os), mis(mis),
+	    maxSctpMessageSize(maxSctpMessageSize), sctpSendBufferSize(sctpSendBufferSize),
+	    isDataChannel(isDataChannel), socket(usrsctp_socket(
+	                                    AF_CONN,
+	                                    SOCK_STREAM,
+	                                    IPPROTO_SCTP,
+	                                    onRecvSctpData,
+	                                    onSendSctpData,
+	                                    SendBufferThreshold,
+	                                    reinterpret_cast<void*>(this->id)))
 	{
 		MS_TRACE();
 
-		// Get a id for this SctpAssociation.
-		this->id = DepUsrSCTP::GetNextSctpAssociationId();
-
 		// Register ourselves in usrsctp.
 		// NOTE: This must be done before calling usrsctp_bind().
 		usrsctp_register_address(reinterpret_cast<void*>(this->id));
 
 		int ret;
 
-		this->socket = usrsctp_socket(
```

Comment:
I'm not big fan of initializing `socket` in above.., we can add a // NOLINT here and bring it back where it was.

Reply:
I agree. If such an initialization of `socket` fails we would not even see the `MS_TRACE`.

Reply:
Done.

I'll add the corresponding // NOLINT directive in a following PR.